sdks/java/io/jdbc/src/test/java/org/apache/beam/sdk/io/jdbc/JdbcUtilTest.java

```diff
@@ -41,4 +53,126 @@ public void testGetPreparedStatementSetCaller() throws Exception {
     String expectedStmt = "INSERT INTO test_table(col1, col2, col3) VALUES(?, ?, ?)";
     assertEquals(expectedStmt, generatedStmt);
   }
+
+  @Test
+  public void testStringPartitioningWithSingleKeyFn() {
+    JdbcReadWithPartitionsHelper<String> helper =
+        JdbcUtil.JdbcReadWithPartitionsHelper.getPartitionsHelper(TypeDescriptors.strings());
+    List<KV<String, String>> expectedRanges =
```

Comment:
@pabloem Unfortunately the string partitions are wrong, for values of type string (of undefined length) a deterministic next value does not exist: `<=  'a'` is simply not the same as `< 'b'`. 

Trying to do this might include values that should be filtered out (e.g. `aa` in the example above) and generate wrong results. 

I'm personally still reluctant to support String partitions for already mentioned reasons. If you want to move forward with strings I would recommend to wrap boundary values in a small wrapper that contains an additional flag to specify if the value is inclusive or not. This way you can adjust your queries accordingly and treat intermediate partitions different from the very last one.

Reply:
oh great catch - I'd like to support strings, but I am fine to leave for a follow up change (maybe use a specialized range object rather than a KV, huh?)

I'll remove the string support for now, and follow up in a different PR.

Reply:
Sounds like the best way forward 👍 Such a range class would also make the interfaces a lot more comprehensive (to eventually expose these to users)